### DISCUSSION TITLE
Unable of setting up Wireguard on RM2

### DISCUSSION THREAD

**Discussion by jpmvferreira** [toltec-dev/toltec#874] [Support] [answered]
Hi there, I'm trying to setup a Wireguard connection between my RM2 and my server. I have Toltec installed and running version 3.3.2.1666 on my RM2.

According to the [Toltec package listing](https://toltec-dev.org/stable/#section-launchers) there is a package named Wireguard. However, trying to installing it in my RM2 reveals
```
# opkg install wireguard
Unknown package 'wireguard'.
Collected errors:
 * opkg_install_cmd: Cannot install package wireguard
```

This is already a little bit weird, but looking in the [Entware repository](https://bin.entware.net/armv7sf-k3.2/Packages.html) reveals that there are two relevant packages: `wireguard-tools` and `wg-quick`. I installed both of them and rebooted my device.

Trying to setup an interface with `wg-quick` reveals the error
```
# wg-quick up wg0                               
Warning: `/opt/etc/wireguard/wg0.conf' is world accessible
[#] ip link add wg0 type wireguard
RTNETLINK answers: Operation not supported
Unable to access interface: Protocol not supported
[#] ip link delete dev wg0
Cannot find device "wg0"
```
and running modprobe reveals
```
# modprobe wireguard
modprobe: FATAL: Module wireguard not found in directory /lib/modules/5.4.70-v1.3.3-rm11x
```

Googling suggested that this could be due to secure boot. If so, I don't know how to solve this in the RM2.

This leaves me with my questions:
1. Why am I not able of installing the `wireguard` package from Toltec's repository in my RM2?
2. Why is the kernel module not loaded after installing the `wireguard-tools` package from Entware repos?

**Comment by Eeems** [comment, marked as answer]
The `wireguard` package is not available after 2.15, as outlined in the following release notes: https://github.com/toltec-dev/toltec/releases/tag/2024-W21. When looking at the package listing, the downloads section only has downloads for `rm1os2` and `rm2os2`.

If you want to use `wireguard` on 3.3.2, you can install the `linux-stracciatella` kernel, and then switch to it using `kernelctl`. It has the wireguard module.

> Googling suggested that this could be due to secure boot.

Secure boot is unrelated to the reMarkable. It's because there is no wireguard module loaded.

> Why is the kernel module not loaded after installing the wireguard-tools package from Entware repos?

Entware is a software repository for network attached storages, routers and other embedded devices. It's not specific to the reMarkable. As such, it doesn't contain kernel modules. As it's targeting multiple devices, some of them will not have the required kernel modules for all the tools to use.

**Reply by jpmvferreira** [reply]
Great, thanks for the reply.
Are there any downsides of using `linux-stracciatella` kernel? Would it be advisable now that I'm changing kernels to use the mainline kernel?

**Reply by Eeems** [reply]
> Great, thanks for the reply. Are there any downsides of using `linux-stracciatella` kernel?

Not that I've found. Just make sure to not reboot if something goes wrong while applying the kernel, as you'll soft-brick.

> Would it be advisable now that I'm changing kernels to use the mainline kernel?

`linux-mainline` is also not available for 3.3.2 yet. When it is available, it has multiple known issues that would make me not recommend it for now.
